modules/publication/ajax/getData.php

```diff
@@ -0,0 +1,305 @@
+<?php
+/**
+ * Publication data retriever
+ *
+ * This retrieves data for publication uploads & editing
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Publication
+ * @author   David <dblader.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+if (isset($_REQUEST['action'])) {
+    $action = $_REQUEST['action'];
+    if ($action === 'getData') {
+        echo json_encode(getData());
+    } elseif ($action === 'getProjectData') {
+        echo json_encode(getProjectData());
+    } else {
+        header("HTTP/1.1 400 Bad Request");
+    }
+}
+/**
+ * Gets publication and parameter_type data from database
+ *
+ * @return array Array of general publication data
+ */
+function getData() : array
+{
+    $db = Database::singleton();
+
+    $data   = array();
+    $titles = $db->pselectCol(
+        'SELECT Title FROM publication',
+        array()
+    );
+
+    // for selecting behavioral variables of interest
+    $bvlVOIs = $db->pselect(
+        "SELECT pt.Name, pt.SourceFrom FROM parameter_type pt ".
+        "JOIN test_names tn ON tn.Test_name=pt.SourceFrom ORDER BY pt.SourceFrom",
+        array()
+    );
+
+    // merge variables and test names into one array
+    $bvlVOIs = array_merge(
+        array_column($bvlVOIs, 'Name'),
+        array_unique(array_column($bvlVOIs, 'SourceFrom'))
+    );
+    sort($bvlVOIs);
+
+    // sets keys and values to be equal
+    $allVOIs['Behavioral'] = array_combine($bvlVOIs, $bvlVOIs);
+
+    // imaging VoIs -- filter out non-human readable DICOM tags
+    $imgVOIs = $db->pselectCol(
+        "SELECT DISTINCT Name FROM parameter_type ".
```

Comment:
on my sandbox, it doesn't appear that imaging variables are properly linked through any of the proper tables. not sure how else to retrieve imaging variables

Reply:
it goes thru the parameter_type_category_rel table.
You need a join between 3 tables

Reply:
need to put data in the table on your sandbox!

Reply:
i'm using CCNA data, resynced last week. not sure what's responsible for linking all these

Reply:
I believe the imaging pipeline scripts are supposed to set the category appropriately when it creates them.

@cmadjar/@MounaSafiHarab  can you confirm?

Reply:
I am pretty sure the MRI pipeline is not populating the `parameter_type_category_rel` table. I think it is only updating the `parameter_type` table. I just checked on my sandbox full of MRI data and the query @davidblader ran also return an empty set for me.

Here is the code that creates the MRI entries in `parameter_type`: https://github.com/aces/Loris-MRI/blob/3bc0d79e46a8039ae363435e6d5471669208ae54/uploadNeuroDB/NeuroDB/File.pm#L479

However, this should be easy to fix in a separate PR (one for the LORIS repo to fix all current projects with imaging data and one on the LORIS-MRI repo so that the parameter type tables get properly populated).

FYI: the same issue has been replicated for the BIDS EEG insertion pipeline since I based the code on the current LORIS-MRI functionality... I was not aware of the `parameter_type_category_rel` table until now...

@driusan Let me know if you want me to create an issue. I am also happy to fix this in separate PRs. :)

Reply:
I think it would be a good idea to create an issue/fix the behaviour on the imaging side, and include the proper query here from the start, then it'll magically fix itself once the imaging side is fixed.

Reply:
@driusan @davidblader PR sent to minor: 
-LORIS: https://github.com/aces/Loris/pull/4263
- LORIS-MRI: https://github.com/aces/Loris-MRI/pull/370

Reply:
@davidblader #4263 has been merged. If you source the following [SQL patch](https://github.com/aces/Loris/blob/minor/SQL/New_patches/2019-01-16_map_parameter_type_and_parameter_type_category_for_imaging.sql) in minor, it should fix your query returning nothing.

Reply:
done